lib/owner-notifier.js

```diff
@@ -13,54 +14,62 @@ class OwnerNotifier {
     };
   }
 
-  check() {
-    return this.getChanges().then(changes => {
-      return this.comment(changes);
-    });
+  async check() {
+    const changes = await this.getChanges();
+
+    return this.comment(await this.getOwnersToPing(changes.owners));
   }
 
-  getChanges() {
-    return this.github.repos.compareCommits(Object.assign({
+  async getChanges() {
+    const compare = await this.github.repos.compareCommits(Object.assign({
       base: this.event.payload.pull_request.base.sha,
       head: this.event.payload.pull_request.head.sha
-    }, this.repo)).then(compare => {
-      return this.getOwners().then(owners => {
-        const paths = compare.files.map(file => file.filename);
-        return new Changes(paths, owners);
-      });
-    });
+    }, this.repo));
+    const owners = await this.getOwners();
+    const paths = compare.files.map(file => file.filename);
+
+    return new Changes(paths, owners);
   }
 
-  getOwners() {
-    const options = Object.assign({path: 'OWNERS'}, this.repo);
-    return this.github.repos.getContent(options).then(data => {
-      return ownersFile(new Buffer(data.content, 'base64').toString());
-    });
+  async getComments() {
+    const options = Object.assign({
+      number: this.event.payload.number
+    }, this.repo);
+
+    return this.github.issues.getComments(options);
   }
 
-  comment(changes) {
-    if (changes.owners.length) {
-      return this.github.issues.createComment(Object.assign({
-        number: this.event.payload.number,
-        body: '/cc ' + changes.owners.join(' ')
-      }, this.repo));
-    }
+  async getOwners() {
+    const options = Object.assign({path: 'OWNERS'}, this.repo);
+    const data = await this.github.repos.getContent(options);
+
+    return ownersFile(new Buffer(data.content, 'base64').toString());
```

Comment:
It only needs an authenticated client for private content. So we could at least start with `context.content(path)` for unauthenticated content and provide some other mechanism for authenticated stuff?